docassemble/AssemblyLine/al_courts.py

```diff
@@ -2,179 +2,218 @@
-          # NOTE: this is causing a problem w/ intrinsic name in the _load_courts method below.
-          pass
-          # self.address.geolocate() # Note that Docassemble has misnamed geocoding to "geolocate"
-          # self.location = self.address.location
-          
+            return str(self.name)
+
+    def short_label_and_address(self) -> str:
+        """
+        Returns a markdown formatted string with the name and address of the court.
+        More concise version without description; suitable for a responsive case.
+        """
+        return f"**{ self.short_label() }**[BR]{ self.address.on_one_line() }"
+
+    def short_description(self) -> str:
+        """
+        Returns a Markdown formatted string that includes the disambiguated name and
+        the description of the court, for inclusion in the results page with radio
+        buttons.
+        """
+        all_info = f"**{ self.short_label() }**"
+        if hasattr(self, "address"):
+            all_info = f"{ all_info }[BR]{ self.address.on_one_line() }"
+        return f"{ all_info }[BR]{ self.description }"
+
+    def from_row(self, df_row, ensure_lat_long=True) -> None:
+        """
+        Loads data from a single Pandas Dataframe into a court object.
+        Note: It will try to convert column names that don't make valid
+        attributes. Best practice is to use good attribute names (no spaces) that don't interfere
+        with existing attributes or methods of DAObject
+        """
+        # A few columns we expect to see:
+        # name
+        # address_address
+        # address_city. Optionally: address_unit, address_zip, address_county, etc. Follow Address object
+        # attribute conventions, but with address_ as prefix.
+        # Optional:
+        # location_latitude and location_latitude will fill the location.latitude/longitude attributes
+        # Other columns will be turned into arbitrary attributes if possible, followed by transforming
+        # underscores.
+        df_row = df_row.dropna()  # Remove any Not a Number entries (null/empty cells)
+        # Handle location specially since we made it on the top level and address object
+        if "location_latitude" in df_row:
+            self.location.latitude = df_row["location_latitude"]
+            self.address.location.latitude = df_row["location_latitude"]
+        if "location_longitude" in df_row:
+            self.location.longitude = df_row["location_longitude"]
+            self.address.location.longitude = df_row["location_longitude"]
+        for attribute_candidate in set(df_row.keys()) - {
+            "location_latitude",
+            "location_longitude",
+        }:
+            if (
+                attribute_candidate.startswith("address_")
+                and attribute_candidate.isidentifier()
+            ):
+                setattr(
+                    self.address, attribute_candidate[8:], df_row[attribute_candidate]
+                )
```

Comment:
This is an .... interesting choice. I hope we'd be able to code around stuff like this, which is crazy unreadable